DietPi-Software | Add Medusa as replacement for SickRage

MichaIng opened the pull request:
**Status**: Testing
- [x] DietPi-Set_Software permissions
- [x] DietPi-Services
- [x] DietPi-Process_tool
- [x] Patch reset install state
- [x] Handle existing SickRage installs (still apply permissions, enable service)? 
- [x] Changelog
- [x] Online docs

**Testing**:
- [x] Jessie
- [x] Stretch
- [x] Buster
- [x] HTTPS
- [x] Update searches and database updates (first login)
- [x] Actual download test

### 🈺 Internal updater without Git:
**Still unsure. On first run Medusa complains unknown version and offers to update, if already on master, and if on older version. It updates by downloading master tarball which works without errors. But afterwards version string remains unknown. So I am not sure if it will recognize, when there is a new update after first run. Manual update check leads to log entry, that no update is required: `Thread_9 :: [918cfe7] No update needed`
- Perhaps it just compares the commit string, which shows up in info page 🤔. This then again would work reliable. User can find out about their version when checking the changelog.**
- [x] Without `libssl-dev`

**Reference**: https://github.com/Fourdee/DietPi/issues/2239

**Commit list/description**:
+ DietPi-Software | Make unrar an own install ID
+ DietPi-Software | Add medusa as replacement for SickRage
+ DietPi-Set_Software | Apply Medusa permissions
+ DietPi-Services | Control Meduse service
+ DietPi-Process_tool | Add Medusa process control
+ DietPi-Process_tool | Minor coding
+ DietPi-Patch | Remove SickRage patches and clear install ID state for Medusa
+ CHANGELOG | Medusa ready for install
+ DietPi-Software | Medusa: Skip libssl-dev install
+ DietPi-Software | rTorrent: Install latest ruTorrent from GitHub automatically
+ DietPi-Software | qBitTorrent: Use default "en" language instead of "en_GB", where several strings and WebUI elements were missing

MichaIng committed:
v6.20

+ DietPi-Software | Make unrar an own install ID
+ DietPi-Software | Add medusa as replacement for SickRage

MichaIng committed:
v6.20

+ DietPi-Set_Software | Apply Medusa permissions

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Apply "dietpi" group to service

MichaIng committed:
v6.20

+ DietPi-Services | Control Meduse service

MichaIng committed:
v6.20

+ DietPi-Process_tool | Add Medusa process control
+ DietPi-Process_tool | Minor coding

MichaIng committed:
v6.20

+ DietPi-Patch | Remove SickRage patches and clear install ID state for Medusa

MichaIng committed:
v6.20

+ CHANGELOG | Medusa ready for install

MichaIng commented:
@Fourdee 
Should we do something about existing SickRage installs?
EG:
- Either keep controlling the service via dietpi-services, or enable the service
- Either keep setting permissions or move install/data dir out of dietpi_userdata

MichaIng committed:
v6.20

+ DietPi-Process_tool | Max nice is 19

Fourdee commented:
@MichaIng 

> Either keep controlling the service via dietpi-services, or enable the service

This one we should keep, ensures if user manually installed it, DietPi would control the services and startup order, and process tool support.

> Either keep setting permissions or move install/data dir out of dietpi_userdata

~We can remove the code for this. No changes to existing user data directory SR data, leave in place.~
Actually, yep, see the issue if user still has this installed.

Hmm, we could patch to move data outside of our user data, but we may need to update the service/settings aswell with new FP?

MichaIng commented:
@Fourdee 
> DietPi would control the services and startup order, and process tool support

Would be actually not too much harm to leave control to systemd, since there are no hard dependencies for the service to start? It should then be properly "Type=Simple", so it will not delay boot.

To avoid the need to keep SickRage related code, we could prompt a whiptail to user to choose new location. The good thing is that we already merged install and data dir all into `userdata/sickrage`, so it's easy to replace this path by the new location then.

Yeah but finally no harm to leave code inside setpermissions/services/process_tool as well. Just inform user that we dropped support and recommend to switch to Medusa, especially if issues appear. Leave uninstall instructions as well.
___________
If you find time, could you do a real download test? I have no indexer account and/or chose the wrong TV shows, that could not be found 😅.
- I tried with qBitTorrent, connecting to it works generally.
- New ruTorrent version works as well (WebUI). I just don't know how to connect to it (or rTorrent directly) from Medusa? Some `action.php` file from ruTorrent is suggested, but that did not work with WebUI login credentials...

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Do not set permissions that are set by "dietpi-set_software setpermissions" anyway. Remove executable flag to systemd unit instead, use default 644.

MichaIng commented:
#### Further checks:
- By default Medusa has no HTTPS enabled.
- This can be done via WebUI or `config.ini`:
```
enable_https = 1
https_cert = server.crt
https_key = server.key
```
- We could do this automatically via DietPi-LetsEncrypt, similar to how this is done for TURN server.
_______
I tested to uninstall `libssl-dev`, since there is not even a build step included in the official install docs. Not sure if this is just a remain from old SickRage docs or such? Service starts fine, no compains in log when purging `libssl-dev`. OpenSSL needs to be installed of course, but is by default on DietPi.
_______
It works well without Git, update checks pass without complains, even an initial update is done successfully:
```
2018-12-26 16:49:42 INFO     CHECKVERSION :: [] Checking for updates using SOURCE
2018-12-26 16:50:32 INFO     Thread_1 :: [] Config backup in progress...
2018-12-26 16:50:32 INFO     Thread_1 :: [] Config backup successful, updating...
2018-12-26 16:50:33 INFO     Thread_1 :: [] Downloading update from u'http://github.com/pymedusa/Medusa/tarball/master'
2018-12-26 16:50:33 INFO     Thread_1 :: [] Clearing update folder u'/mnt/dietpi_userdata/medusa/sr-update' before extracting
2018-12-26 16:50:47 INFO     Thread_1 :: [] Extracting file /mnt/dietpi_userdata/medusa/sr-update/sr-update.tar
2018-12-26 16:50:59 INFO     Thread_1 :: [] Deleting file /mnt/dietpi_userdata/medusa/sr-update/sr-update.tar
2018-12-26 16:50:59 INFO     Thread_1 :: [] Moving files from /mnt/dietpi_userdata/medusa/sr-update/pymedusa-Medusa-918cfe7 to /mnt/dietpi_userdata/medusa
...
2018-12-26 16:51:29 INFO     CHECKVERSION :: [918cfe7] Checking for updates using SOURCE
2018-12-26 16:56:26 INFO     Thread_2 :: [918cfe7] Checking for updates using SOURCE
2018-12-26 16:56:27 INFO     Thread_2 :: [918cfe7] No update needed
```
But switching branch and "checkout" of course does not work. I will test internal updater by manually installing an older version from releases. But for sure, full functionality is only available with Git. Even the info page does not show a version string:

Medusa Info: | Branch: master Commit: 918cfe7ce01808499585ffe6c1356fd5ed369f76 Version: Unknown Database: 44.12
-- | --
Python Version: | 2.7.9 (default, Sep 25 2018, 20:42:16) [GCC 4.9.2]
SSL Version: | OpenSSL 1.0.1t 3 May 2016
OS: | Linux-3.16.0-7-amd64-x86_64-with-debian-8.11
Locale: | en_GB.UTF-8
- 🈺 **Still unsure. On first run Medusa complains unknown version and offers to update, if already on master, or older version. It updates by downloading master tarball which works without errors. But afterwards version string is still unknown. So I am not sure if it will recognize, when there is a new update after first run.** 
__________
#### Python3 support for Buster:
From WebUI changelog:

> ### Unreleased
> #### New Features
> - Added support for Python 3 (>= 3.5.0) (#4982)
> #### Improvements
> #### Fixes

- Merged into develop 4 days ago: https://github.com/pymedusa/Medusa/pull/4982

Interestingly the start script already states to support Python3 🤔:
```
    if sys.version_info.major == 3 and sys.version_info.minor < 5:
        print('Medusa supports Python 2 from version 2.7.10 and Python 3 from version 3.5.0, exiting!')
        raise Exception('Incorrect Python version. Shutting down!')
    main()
```

Release schedule is around 3 weeks, so this should be in master/release fast, no need to take special care about Buster, install Python2.7 or such things.
**€: 🈯️ Ah okay, even on Buster, default Python version is still 2.7 😃.**

MichaIng committed:
v6.20

+ DietPi-Software | Tiny blank

MichaIng committed:
v6.20

+ DietPi-Software | Medusa: Skip libssl-dev install
+ DietPi-Software | rTorrent: Install latest ruTorrent from GitHub automatically

MichaIng committed:
v6.20

+ DietPi-Software | qBitTorrent: Use default "en" language instead of "en_GB", where several strings and WebUI elements were missing

Fourdee committed:
Merge branch 'dev' into medusa

MichaIng committed:
v6.20

+ DietPi-Process_tool | Re-add SickRage process handling to support existing installs

MichaIng committed:
v6.20

+ DietPi-Services | Re-add SickRage service handling keep supporting existing installs

MichaIng committed:
v6.20

+ DietPi-Set_Software| setpermissions: Re-add SickRage permissions to support existing installs

MichaIng committed:
v6.20

+ DietPi-Patch | Inform user about SickRage being replaced by Medusa

MichaIng committed:
META | README: Add Medusa GitHub link

Fourdee commented on dietpi/patch_file:
```diff
@@ -1551,6 +1517,17 @@ Pin-Priority: -1' > /etc/apt/preferences.d/dietpi-docker_fix
 			[[ -f /etc/apache2/sites-available/dietpi-nextcloud.conf ]] && a2ensite dietpi-nextcloud
 			#-------------------------------------------------------------------------------
+			#Clear install state for Medusa
+			(( $G_DIETPI_VERSION_RC == 0 && $G_DIETPI_INSTALL_STAGE == 2 )) && G_CONFIG_INJECT 'aSOFTWARE_INSTALL_STATE\[116\]=' 'aSOFTWARE_INSTALL_STATE[116]=0' /DietPi/dietpi/.installed
```
@MichaIng 

Don't believe we can use ```G_DIETPI_VERSION_RC ``` here? EG:

- v6.19.7
- Update to v6.20.2
- ```G_DIETPI_VERSION_RC ``` is taken from server version (2)
- ```G_DIETPI_VERSION_RC == 0``` never triggered.

Might be easier to simply?
```
if [[ ! -d $G_FP_DIETPI_USERDATA/medusa ]]; then

 G_CONFIG_INJECT 'aSOFTWARE_INSTALL_STATE\[116\]='  'aSOFTWARE_INSTALL_STATE[116]=0' /DietPi/dietpi/.installed

fi
```

MichaIng replied:
@Fourdee 
Thought to add this for beta testers. If they test Medusa, it's install state should not be reverted on every update, so just do this once on initial beta?

But yeah you are right, RC == 0 is only the case, AFTER first beta release (if we leave RC version == 0).

~Check `/DietPi/dietpi/.version` for line 3 `G_DIETPI_VERSION_SUB<=19`?~ €: Your solution is easiest!

MichaIng committed:
v6.20

+ DietPi-Patch | Fix Medusa install detection when clearing it's install state

MichaIng committed:
v6.20

+ DietPi-Software | Minor: White space + forgotten Download_Install() case

Fourdee commented:
@MichaIng 

Looking good, thank you, lets merge for testing 👍

MichaIng commented:
@Fourdee 
Quick note: Unrar is in developer tools category, because there is non that fit's really good. We could open a new one "Archiver" or something like that, but that would only have this one entry since we have all other mentionable ones pre-installed 🤔.

Fourdee commented:
@MichaIng 

> Quick note: Unrar is in developer tools category, because there is non that fit's really good. We could open a new one "Archiver" or something like that, but that would only have this one entry since we have all other mentionable ones pre-installed 

Yep, unless we offer 2 or more compression based installations, leave it where it is for now?

MichaIng commented:
@Fourdee 
Agree
